### PR TITLE
Add OPC_MODE

### PR DESCRIPTION
**Pull request by MartinDaCosta53**

### CODE REVIEW

File: src/EventTeachingService.cpp [hunk 1]
```diff
@@ -32,10 +48,7 @@ Processed EventTeachingService::handleMessage(unsigned int opc, CANFrame *msg)
 
     if (nn == module_config->nodeNum)
     {
-      bLearn = true;
-      // DEBUG_SERIAL << F("> set lean mode ok") << endl;
-      // set bit 5 in parameter 8
-      bitSet(controller->_mparams[8], 5);
+     enableLearn();
```
Comment (SvenRosvall) [inline src/EventTeachingService.cpp:51]: Please put one more space in here so you have a nice indentation.

File: src/MinimumNodeService.cpp [hunk 2]
```diff
@@ -16,16 +18,24 @@ void MinimumNodeService::setController(Controller *cntrl)
...
+  //Initialise instantMode
+  instantMode = module_config->currentMode;
+  //DEBUG_SERIAL << F("> instant MODE initialise as: ") << instantMode << endl;
+}
+
 //
 /// initiate the transition from Uninitialised to Normal mode
 //
-void MinimumNodeService::initNormal()
+void MinimumNodeService::initSetup()
```
Comment (SvenRosvall) [inline src/MinimumNodeService.cpp:31]: Yes. Much better name!

File: src/MinimumNodeService.cpp [hunk 3]
```diff
@@ -329,7 +345,9 @@ Processed MinimumNodeService::handleMessage(unsigned int opc, CANFrame *msg)
           else
           {
             // Couldn't find the service.
-            controller->sendGRSP(OPC_RQSD, getServiceID(), CMDERR_INV_PARAM_IDX);
+            controller->sendCMDERR(CMDERR_INV_EN_IDX);
+            // NOTE: error code 9 is really for parameters. But there isn't any better for CMDERR.
+            controller->sendGRSP(OPC_RQSD, getServiceID(), GRSP_INVALID_SERVICE);
```
Comment (SvenRosvall) [inline src/MinimumNodeService.cpp:350]: Hmm, not sure what happened here. This is undoing my recent change. Please keep the red code and remove the green code.

File: src/EventTeachingService.cpp (hunk 1 above)
Comment (MartinDaCosta53) [reply src/EventTeachingService.cpp:51]: OK

File: src/MinimumNodeService.cpp [hunk 4]
```diff
@@ -343,6 +361,97 @@ Processed MinimumNodeService::handleMessage(unsigned int opc, CANFrame *msg)
...
+            controller->requestedMode(newMode);
+            instantMode = newMode;
+          } 
+          
+          break;
+            
+        }
+      }
       
       return PROCESSED;
```
Comment (SvenRosvall) [inline src/MinimumNodeService.cpp:453]: Remove this return statement.

**Pull request merged**